test/src/test/java/org/corfudb/runtime/view/SequencerViewTest.java

```diff
@@ -21,25 +25,42 @@
     @Test
     public void canAcquireFirstToken() {
         CorfuRuntime r = getDefaultRuntime();
-        assertThat(r.getSequencerView().nextToken(Collections.emptySet(), 1).getToken())
+        assertThat(r.getSequencerView().nextToken().getToken())
                 .isEqualTo(new Token(0L, 0L));
     }
 
+    @Test
+    public void canQueryMultipleStreams() {
+        CorfuRuntime r = getDefaultRuntime();
+
+        UUID stream1 = UUID.randomUUID();
+        UUID stream2 = UUID.randomUUID();
+        UUID stream3 = UUID.randomUUID();
+
+        assertThat(r.getSequencerView().nextToken(stream1).getToken())
+                .isEqualTo(new Token(0l, 0l));
+        assertThat(r.getSequencerView().nextToken(stream2).getToken())
+                .isEqualTo(new Token(1l, 0l));
+
+        assertThat(r.getSequencerView().queryTails(stream1, stream2, stream3).getStreamTails())
```

Comment:
RE the point mentioned above, if we just pass stream1 to queryTails, return size would be zero.

Reply:
If you query a single tail, then you get the result in token, if you query multiple streams then the response is stored in streamTails and the token stores the global tail.